### ISSUE TITLE
Using <div> with inline HTML causes incorrect parsing

### ISSUE THREAD

**Issue by ron-at-swgy** [kristapsdz/lowdown#127]
I have run into an issue where inline HTML to indicate divs (for a 3 column layout in this case) causes the parser to see the rest of the file as a single HTML block.

Consider the following fragment of a markdown file:

```markdown
# 504th PIR Association Member Information


## Welcome Members!


<div class="container">

<div class="column column-1">

### Reference information


- Officers
- Directors
- SOPs (roster update, roster query)


</div>
<div class="column column-2">

... more content elided ...
</div>
</div>
```

Invoking lowdown with `-thtml --html-no-skiphtml --html-no-escapehtml` provides an HTML file with processing stopped at the first div:

```html
<h1 id="504th-pir-association-member-information">504th PIR Association Member Information</h1>

<h2 id="welcome-members">Welcome Members!</h2>

<div class="container">

<div class="column column-1">

### Reference information


- Officers
- Directors
- SOPs (roster update, roster query)


</div>
<div class="column column-2">
 ... more content elided ...
```

As you can see, Markdown to HTML translation ceased at the first div marker. If I add an hr for testing sake, it does not stop processing. If I remove the divs, processing continues. 

I reviewed #37 and believe this is a distinct issue. For reference, here is the output of -Ttree:

```
[rdahlgren@builder :: ~/Development/504] $ lowdown -Ttree for-members.md
LOWDOWN_ROOT (0)
  LOWDOWN_DOC_HEADER (1)
  LOWDOWN_HEADER (2)
    level: 0
    LOWDOWN_NORMAL_TEXT (3)
      data: 40 Bytes: 504th PIR Associatio...
  LOWDOWN_HEADER (4)
    level: 1
    LOWDOWN_NORMAL_TEXT (5)
      data: 16 Bytes: Welcome Members!
  LOWDOWN_BLOCKHTML (6)
    data: 332 Bytes: <div class="containe...
```

I have used inline HTML included divs in other cases with lowdown with no problem. The difference is these other cases were "self-contained" - meaning the entire content of the div was present before the next line of markdown.

How can I help?

**Comment by kristapsdz** [comment]
This is the default behaviour of the original Markdown specification. From pandoc(1):

```
     Original Markdown allows you to include HTML “blocks”: blocks of HTML
     between balanced tags that are separated from the surrounding text with
     blank lines, and start and end at the left margin.  Within these blocks,
     everything is interpreted as HTML, not Markdown; so (for example), * does
     not signify emphasis.
```

However, some renderers (including pandoc), extend this:

```
     Pandoc behaves this way when the markdown_strict format is used; but by
     default, pandoc interprets material between HTML block tags as Markdown.
```

Implementing something like this isn't easy, but it's not impossible: it would require breaking apart the text in a `LOWDOWN_BLOCKHTML` into a series of `LOWDOWN_RAW_HTML` separated by parsed markdown.  If you'd like to take a look at this, it would require tokenising the interval of the block and calling the parser for non-HTML subintervals.  See document.c where `LOWDOWN_BLOCKHTML` is pushed onto the parse stack.

**Comment by ron-at-swgy** [comment]
I understand, thank you. I may be able to do some CSS tricks with the "p" element to get a two-column layout on wider display ports in the interim. I'll close this issue.

**Comment by ron-at-swgy** [comment]
Kristaps, I'm don't understand the difference between node types `LOWDOWN_BLOCKHTML` and `LOWDOWN_RAW_HTML`. Would it make sense to parse my case into a few `LOWDOWN_BLOCKHTML` nodes with other markdown nodes in between? I see that `<hr />` and `<br />` are parsed as `LOWDOWN_BLOCKHTML`. When would I expect to see `LOWDOWN_RAW_HTML`?

Thank you,
Ron

**Comment by ron-at-swgy** [comment]
Addressed in #128
